src/cloud/metrics/controllers/server.go

```diff
@@ -85,11 +87,21 @@ func (s *Server) Start() {
 		log.WithError(err).Fatal("Failed to get table from BigQuery")
 	}
 
+	t := &bq.Table{
+		Schema: schema,
+		Table:  table,
+	}
+
 	for _, shard := range vzshard.GenerateShardRange() {
-		bqWriteChan := make(chan []*bigquery.StructSaver, 2048)
+		bqWriteChan := make(chan interface{}, 4096)
+		bqBatchInserter := bq.BatchInserter{
+			BatchSize:   1024,
+			PushTimeout: 10 * time.Second,
```

Comment:
I would probably make the push timeout a bit longer. I think its fine to give up latency for less calls to bigquery.Put